addons/binding/org.openhab.binding.xiaomivacuum/src/main/java/org/openhab/binding/xiaomivacuum/discovery/VacuumDiscovery.java

```diff
@@ -0,0 +1,173 @@
+                        "No token discovered for device {}. To discover token reset the vacuum & connect to it's wireless network and re-run discovery");
+                thingDiscovered(DiscoveryResultBuilder.create(uid)
+                        .withProperty(XiaomiVacuumBindingConstants.PROPERTY_HOST_IP, i.getKey())
+                        .withLabel("Xiaomi Robot Vacuum").build());
+            } else {
+                logger.debug("Discovered token for device {}: {} ('{}')", id, token, new String(msg.getChecksum()));
+                thingDiscovered(DiscoveryResultBuilder.create(uid)
+                        .withProperty(XiaomiVacuumBindingConstants.PROPERTY_HOST_IP, i.getKey())
+                        .withProperty(XiaomiVacuumBindingConstants.PROPERTY_TOKEN, token)
+                        .withLabel("Xiaomi Robot Vacuum").build());
+            }
+        }
+    }
+
+    /**
+     * @return broadcast addresses for all interfaces
+     */
+    private static TreeSet<String> getBroadcastAddresses() {
+        TreeSet<String> broadcastAddresses = new TreeSet<String>();
+        try {
+            broadcastAddresses.add("224.0.0.1");
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                try {
+                    NetworkInterface networkInterface = interfaces.nextElement();
+                    if (networkInterface.isLoopback() || !networkInterface.isUp()) {
+                        continue;
+                    }
+                    for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                        String address = interfaceAddress.getBroadcast().getHostAddress();
+                        if (address != null) {
+                            broadcastAddresses.add(address);
+                        }
+                    }
+                } catch (Exception e) {
+                    // ignore
+                }
+            }
+        } catch (Exception e) {
+            logger.debug("Error collecting broadcast addresses: {}", e.getMessage(), e);
+        }
+        return broadcastAddresses;
+    }
+
+    public static HashMap<String, byte[]> sendDiscoveryRequest(String ipAddress) {
+        HashMap<String, byte[]> responses = new HashMap<String, byte[]>();
+
+        try (DatagramSocket clientSocket = new DatagramSocket()) {
+            clientSocket.setReuseAddress(true);
+            clientSocket.setBroadcast(true);
+            clientSocket.setSoTimeout(TIMEOUT);
+            byte[] sendData = XiaomiVacuumBindingConstants.DISCOVER_STRING;
+            DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length, InetAddress.getByName(ipAddress),
+                    PORT);
+            for (int i = 1; i < 3; i++) {
+                clientSocket.send(sendPacket);
+            }
+            sendPacket.setData(new byte[1024]);
+            while (true) {
```

Comment:
The loop is ended by the timeout. 
This allows for multiple vacuums to present themselves during discovery rather than returning just the first one